src/main/java/UserInterface.java

```diff
@@ -0,0 +1,223 @@
+         * Strings in input have been read.
+         */
+        ArrayList<StringBuilder> taskParts = new ArrayList<>();
+        StringBuilder prepositions = new StringBuilder();
+        for (int taskPartsIterator = 0, inputIterator = 0; inputIterator < input.size(); ++taskPartsIterator) {
+            String finalWord = input.get(inputIterator);
+            if (finalWord.startsWith("/")) {
+                prepositions.append(finalWord);
+            }
+            inputIterator++;
+            taskParts.add(new StringBuilder());
+            for (; inputIterator < input.size() && !input.get(inputIterator).startsWith("/"); ++inputIterator) {
+                taskParts.get(taskPartsIterator).append(input.get(inputIterator)).append(" ");
+            }
+        }
+        taskParts.add(prepositions);
+        return taskParts;
+    }
+
+    /**
+     * Prints all tasks stored in memory by addTask(Task) to stdout
+     */
+    private void listTasks() {
+        printDivider();
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 1; i <= this.tasks.size(); i++) {
+            System.out.println(i + "." + this.tasks.get(i - 1));
+        }
+        printDivider();
+    }
+
+    /**
+     * Marks the task selected by the user as done. Tasks are selected by their visual index on the list
+     * (starting from 1, not 0) and not by name.
+     * @param task The index of the task to be marked done.
+     */
+    private void doTask(String task) {
+        if (task == null) {
+            System.out.println(("Error: Task to be done is null."));
+            return;
+        }
+        try {
+            this.tasks.get(Integer.parseInt(task) - 1).doTask();
+            printDivider();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(this.tasks.get(Integer.parseInt(task) - 1));
+            printDivider();
+        } catch (Exception e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Marks the task selected by the user as undone. Tasks are selected by their visual index on the list
+     * (starting from 1, not 0) and not by name.
+     * @param task The index of the task to be marked done.
+     */
+    private void undoTask(String task) {
+        if (task == null) {
+            System.out.println("Error: Task to be undone is null.");
+            return;
+        }
+        try {
+            this.tasks.get(Integer.parseInt(task) - 1).undoTask();
+            printDivider();
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println(this.tasks.get(Integer.parseInt(task) - 1));
+            printDivider();
+        } catch (Exception e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Prints a goodbye message and returns.
+     */
+    private void printGoodbye() {
```

Comment:
Might want to consider saving the system.out.println code as a function to ensure SLAP in the printGoodbye function

Reply:
Thank you. I will consider this.